buildpack/fetcher_test.go

```diff
@@ -49,9 +49,26 @@ func testBuildpackFetcher(t *testing.T, when spec.G, it spec.S) {
 			h.AssertEq(t, out.Version, "some-buildpack-version")
 			h.AssertEq(t, out.Stacks[0].ID, "some.stack.id")
 			h.AssertEq(t, out.Stacks[1].ID, "other.stack.id")
-			h.AssertNotEq(t, out.Dir, "")
-			h.AssertDirContainsFileWithContents(t, out.Dir, "bin/detect", "I come from a directory")
-			h.AssertDirContainsFileWithContents(t, out.Dir, "bin/build", "I come from a directory")
+			h.AssertNotEq(t, out.Path, "")
+			h.AssertDirContainsFileWithContents(t, out.Path, "bin/detect", "I come from a directory")
+			h.AssertDirContainsFileWithContents(t, out.Path, "bin/build", "I come from a directory")
+		})
+
+		it("fetches a buildpack from a tgz", func() {
+			downloadPath := filepath.Join("testdata", "buildpack.tgz")
+			mockDownloader.EXPECT().
+				Download(downloadPath).
+				Return(downloadPath, nil)
+
+			out, err := subject.FetchBuildpack(downloadPath)
+			h.AssertNil(t, err)
+			h.AssertEq(t, out.ID, "bp.one")
+			h.AssertEq(t, out.Version, "some-buildpack-version")
+			h.AssertEq(t, out.Stacks[0].ID, "some.stack.id")
+			h.AssertEq(t, out.Stacks[1].ID, "other.stack.id")
+			h.AssertNotEq(t, out.Path, "")
+			h.AssertOnTarEntry(t, out.Path, "./bin/detect", h.ContentEquals("I come from a directory"))
```

Comment:
The contents of this test string are a bit confusing in the tgz test case context.